event/event.go

```diff
@@ -80,6 +104,44 @@ func NewRawSubscription(url string, maxConcurrency int) (*RawSubscription, error
 	}, nil
 }
 
+// Serve will start serving all events from the subscription calling handler for each
+// event. It will run until [Subscription.Shutdown] is called.
+// If the error is nil Ack is sent.
+// If a non-nil error is returned by the handler Unack will be sent.
+// If a received event is not a valid JSON it will be discarded as malformed and a Nack will be sent automatically.
+// If a received event has the wrong name it will be discarded as malformed and a Nack will be sent automatically.
+// Serve may be called multiple times, each time will start a new serving service that will
+// run up to "maxConcurrency" goroutines.
+func (s *Subscription[T]) Serve(handler EventHandler[T]) error {
+	return s.rawsub.Serve(func(msg []byte) error {
+		var event Envelope[T]
+
+		if err := json.Unmarshal(msg, &event); err != nil {
+			return fmt.Errorf("parsing event as JSON, event: %v, error: %v", string(msg), err)
+		}
+
+		if event.Name != s.name {
+			return fmt.Errorf("event name doesn't match %q: event: %v", s.name, string(msg))
+		}
```

Comment:
This is a publisher error, right? For such case, we're nacking and sending to the DLQ?